rest/route/build_test.go

```diff
@@ -56,9 +76,21 @@ func (s *BuildByIdSuite) TestFindByIdProjFound() {
 	s.True(ok)
 	s.Equal(utility.ToStringPtr("build1"), b.Id)
 	s.Equal(utility.ToStringPtr("branch"), b.ProjectId)
+	s.Equal("task1", b.TaskCache[0].Id)
+
+	s.rm.(*buildGetHandler).buildId = "build2"
+	resp = s.rm.Run(context.TODO())
+	s.Equal(resp.Status(), http.StatusOK)
+	s.NotNil(resp.Data())
+
+	b, ok = (resp.Data()).(*model.APIBuild)
+	s.True(ok)
+	s.Equal(utility.ToStringPtr("build2"), b.Id)
+	s.Equal(utility.ToStringPtr("notbranch"), b.ProjectId)
+	s.Equal("task2", b.TaskCache[0].Id)
```

Comment:
Can we also validate that something else that would be added via SetFromTaskCache is added here? (since this would be true even if we didn't find the task from the find query)